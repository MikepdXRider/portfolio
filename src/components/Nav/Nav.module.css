.nav {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 3vw;
}

.link {
  color: darkslategray;
  font-size: 20px;
  text-decoration: none;
}
