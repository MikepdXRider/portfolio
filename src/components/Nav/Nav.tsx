import React from 'react';
import styles from './Nav.module.css';

interface NavDataItem {
  textContent: string;
  href: string;
  ariaLabel: string;
}

interface NavProps {
  navData: NavDataItem[];
}

export default function Nav({ navData }: NavProps) {
  return (
    <nav className={styles.nav}>
      {navData.map((linkData) => (
        <a
          className={styles.link}
          href={linkData.href}
          aria-label={linkData.ariaLabel}
        >
          {linkData.textContent}
        </a>
      ))}
    </nav>
  );
}
