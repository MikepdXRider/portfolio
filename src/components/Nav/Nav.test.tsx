// testing
import { render } from '@testing-library/react';
// components
import Nav from './Nav';

describe('Nav component', () => {
  const mockNavData = [
    {
      textContent: 'mock text content 1',
      href: 'mock href 1',
      ariaLabel: 'mock aria label 1',
    },
    {
      textContent: 'mock text content 2',
      href: 'mock href 2',
      ariaLabel: 'mock aria label 2',
    },
  ];

  it('matches the existing snapshot', () => {
    const { container } = render(<Nav navData={mockNavData} />);

    expect(container).toMatchSnapshot();
  });
});
