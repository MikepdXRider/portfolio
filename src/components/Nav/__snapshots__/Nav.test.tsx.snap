// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches the existing snapshot 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      aria-label="mock aria label 1"
      class="link"
      href="mock href 1"
    >
      mock text content 1
    </a>
    <a
      aria-label="mock aria label 2"
      class="link"
      href="mock href 2"
    >
      mock text content 2
    </a>
  </nav>
</div>
`;
